src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBinderFactory.cs

```diff
@@ -73,6 +73,11 @@ public IModelBinder CreateBinder(ModelBinderFactoryContext context)
 
         private IModelBinder CreateBinderCore(DefaultModelBinderProviderContext providerContext, object token)
         {
+            if (!providerContext.Metadata.IsBindingAllowed)
+            {
+                return NoOpBinder.Instance;
+            }
```

Comment:
Need tests to make sure `[BindNever]` is actually working. Suggest something in the integration tests project.